app/services/mentions/create_all.rb

```diff
@@ -10,32 +10,55 @@ def self.call(*args)
 
     def call
       # Only works for comments right now.
+      mentioned_users = users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+
+      delete_mentions_removed_from_notifiable_text(mentioned_users)
+      create_mentions_for(mentioned_users)
+    end
+
+    private
+
+    def users_mentioned_in_notifiable_text_that_are_existing_and_not_its_owners
+      mentioned_usernames = extract_usernames_from_mentions_in_text
+
+      collect_existing_users(mentioned_usernames).
+        then do |existing_mentioned_users|
```

Comment:
Oh wow, TIL about `then` in Ruby! I'd only ever heard of `yield_self` :)

Reply:
I find the `then` alias a bit confusing and less intention-revealing than `yield_self`.

Reply:
I don't mind `yield_self`, so I switched to that.